dmriprep/utils/vectors.py

```diff
@@ -0,0 +1,295 @@
+from __future__ import division, print_function
+import os
+import nibabel as nib
+import numpy as np
+
+
+def make_gtab(fbval, fbvec, vector_dir, final, b0_thr=50):
+    """
+    Create gradient table from bval/bvec.
+
+    Parameters
+    ----------
+    fbval : str
+        File path of the b-values.
+    fbvec : str
+        File path of the b-vectors.
+    vector_dir : str
+        Path to vectors directory in subject output.
+    final : bool
+        Indicates whether to flag the output gradient
+    b0_thr : float
+        Gradient threshold below which volumes and vectors are considered B0's.
+
+    Returns
+    -------
+    gtab_file : str
+        File path to pickled DiPy gradient table object.
+    gtab : Obj
+        DiPy object storing diffusion gradient information.
+    final_fbval : str
+        File path of saved final b-values if `final` is True.
+    final_fbvec : str
+        File path of saved final b-vectors if `final` is True.
+    """
+    import os
+    from dipy.io import save_pickle, read_bvals_bvecs
+    from dipy.core.gradients import gradient_table
+
+    # Read b-values and b-vectors from disk
+    # Files can be either ‘.bvals’/’.bvecs’ or ‘.txt’ or ‘.npy’ (containing
+    # arrays stored with the appropriate values).
+    if fbval and fbvec:
+        bvals, bvecs = read_bvals_bvecs(fbval, fbvec)
+    else:
+        raise OSError("Either bvals or bvecs files not found (or rescaling failed)!")
+
+    # Creating the gradient table
+    gtab = gradient_table(bvals, bvecs, b0_threshold=b0_thr)
+
+    # Show info
+    print(gtab.info)
+
+    # Save bval/bvec text file equivalents in the case that these are the final
+    # versions of our gradient vectors
+    if final is True:
+        final_bval_path = "%s%s" % (vector_dir, '/final_bval.bval')
+        final_bvec_path = "%s%s" % (vector_dir, '/final_bvec.bvec')
+        np.savetxt(final_bval_path, bvals.astype('int'), fmt='%d', newline=' ')
+        np.savetxt(final_bvec_path, bvecs.T.astype('float'), fmt='%8f')
+        # Save gradient table to pickle
+        gtab_file = "%s%s" % (vector_dir, "/gtab_final.pkl")
+        save_pickle(gtab_file, gtab)
+    else:
+        def next_path(path_pattern):
+            i = 1
+            while os.path.exists(path_pattern % i):
+                i = i * 2
+            a, b = (i // 2, i)
+            while a + 1 < b:
+                c = (a + b) // 2 # interval midpoint
+                a, b = (c, b) if os.path.exists(path_pattern % c) else (a, c)
+            return path_pattern % b
+        # Save gradient table to pickle
+        gtab_file = next_path(vector_dir + "/gtab_%s.pkl")
+        save_pickle(gtab_file, gtab)
+        final_bval_path = None
+        final_bvec_path = None
+
+    return gtab_file, gtab, final_bval_path, final_bvec_path
+
+
+def rescale_bvec(bvec):
+    """
+    Normalizes b-vectors to be of unit length for the non-zero b-values. If the
```

Comment:
This should be in imperative mood:

```suggestion
    Normalize b-vectors to be of unit length for the non-zero b-values.

    If the
```